youtube_dl/extractor/egghead.py

```diff
@@ -82,3 +79,29 @@ def _real_extract(self, url):
             'view_count': int_or_none(lesson.get('plays_count')),
             'tags': try_get(lesson, lambda x: x['tag_list'], list),
         }
+
+        def find_id_and_dlurl():
+            vid_id = lesson.get('wistia_id')
+            if vid_id:
+                return {'ie_key': 'Wistia',
+                        'id': vid_id, 'url': 'wistia:' + vid_id}
+
+            self.report_warning('Cannot find an proper ID, will use lesson name URL slug')
+            vid_id = self._html_search_regex(
+                r'^https?://egghead\.io/lessons/([A-Za-z0-9][A-Za-z0-9-]*)$',
+                lesson.get('http_url'),
+                'lesson name URL part as ID of last resort',
+                group=1)
+
+            mu = lesson.get('media_urls')
```

Comment:
I'm literally going to keep changing these around to different properties as soon as this PR is merged.